src/main/java/org/primeframework/mvc/cors/CORSFilter.java

```diff
@@ -280,10 +280,10 @@ private CORSRequestType checkRequestType(HTTPRequest request, String origin) {
       } else if (HTTPMethod.GET.is(method) || HTTPMethod.HEAD.is(method)) {
         requestType = CORSRequestType.SIMPLE;
       } else if (HTTPMethod.POST.is(method)) {
+        Long contentLength = request.getContentLength();
         String contentType = request.getContentType();
-        if (contentType != null) {
-          contentType = contentType.toLowerCase().trim();
-          if (SimpleHTTPRequestContentTypes.contains(contentType)) {
+        if (contentType != null || 0 == contentLength) {
```

Comment:
I don' think checking the `Content-Length` header alone is adequate to know if a body is present.

If you the caller is chunking the request the `Content-Length` will be omitted, and in place of it will be the `Transfer-Encoding` header. 

So I think if we were to use this logic, we would allow a `POST` request in this path if `Content-Length` & `Transfer-Encoding` are both `null`. 

Or, in your testing, are you seeing a `Content-Length: 0` header on a `POST` request w/out a body? 

My only concern with making this change in general is that if all CORS filter use this logic, changing it here will only fix our React SDK when used with FusionAuth, but it won't necessarily work with any other backend.

So we should see if we can confirm how pervasive this type of validation is - if it is part of the spec, or was just the opinion of the writer of this filter. If it is pervasive, then perhaps the better (hacky) way to fix it would be to just set the `Content-Type` header to `text/plain;`. 

The two Java filters I found were from eBay and Apache Tomcat and both have this logic. But I was unable to find this specific requirement as part of the CORS w3 spec, so perhaps it is implied, or just one interpretation of the spec.

Reply:
I can't now recreate the problem I was having where the content-type header was being ignored if the body was empty, so assuming the problem was me I am going to close this. The dummy content-type header still feels kind of weird in this situation but does feel better than modifying the cors filter if we aren't sure.